src/index.js

```diff
@@ -20,12 +20,32 @@ function deleteSingleTask() {
   });
 }
 
+function clear() {
+  const clear = document.getElementById('clear-button');
+  clear.addEventListener('click', () => {
+    const checkboxes = document.getElementsByClassName('task');
+    const checkboxesArr = Array.from(checkboxes);
+
+    let deleteList = [];
+
+    checkboxesArr.forEach((el, i) => {
+      if (el.classList.contains('task-done')) deleteList.push(i);
+    });
```

Comment:
- [ ] As you can see in the screenshot below you were asked to use `Array.filter` method to clear All completed tasks from your To Do List, Kindly refactor this part to use this filter method as it's simple to use and good way for best practice.

**Syntax :** 

```JS
const newArray= arr.filter((task)=> task.completed);
```

![contactReel](https://user-images.githubusercontent.com/80895497/154805836-4595af0d-e28c-4efb-a011-18d0db9612ad.png)

Reply:
Thanks for letting me know, but there's some kind of overlap that doesn't let me: 

Whenever I use filter, the code breaks (I suppose it has to do with the time filter takes to complete) everything after filter executes, even in the original function that called it, but it doesn't end there. (Again, I assume this is because filter takes too long) The function that called the one filter is in executes again, and again, maybe 6 or 10 times, before it just doesn't.
Filter works, it does save the correct values in local storage, but everything else breaks in the process.
I'm fully available today if you could help me.